HW #2

drag0nfet opened the pull request:
- голый функционал реестра, но getByName работает с оговоркой, подробности в комменте

- задержка из-за болезни <3

drag0nfet committed:
hw#2 - 1st commit
- голый функционал реестра, но getByName работает с оговоркой, подробности в комменте

lefasdayan commented on ru/turing/courses/lesson3/Registry.java:
```diff
@@ -1,13 +1,49 @@
-    private final Map<Object, Object> storage = new HashMap<>(); //todo поменять Object, Object на то, что будет у вашего класса в примере
-
-    //todo написать и реализовать методы
+public abstract class Registry <K extends Number, String>{
```
Во-первых, не хватает условного Main, в котором ты бы показал на практике, как работаешь с регистром. А еще, если бы был мейн, ты бы понял, что класс в таком виде не сможешь использовать, потому что он абстрактный.

Во-вторых, по заданию нужно в качестве value засовывать объекты реальные, а не просто String) Можешь взять классы из прошлой домашки и использовать их

drag0nfet committed:
HW#2 - 2nd commit
- изменил прошлый класс Animal, добавил equals и hashCode
- Registry - абстрактный класс без реализации
- AnimalRegistry - неабстрактный класс, функционал расширил наличием имени регистра и функциями 1. на поиск по имени животного и 2. на вывод всего содержимого
- добавил класс Solution3 для реализации класса AnimalRegistry

p.s. В планах больше не пропадать, вроде большинство долгов теперь закрыто и здоровье нормализовалось

lefasdayan commented:
Отлично, молодец)
